sherpa/astro/tests/test_astro_data2.py

```diff
@@ -85,6 +85,68 @@ def test_need_numpy_channels():
     assert pha.get_filter() == 'No noticed bins'
 
 
+@pytest.mark.parametrize("chtype,expected,args",
+                         [("channel", '1:10', []),
+                          ("channel", '', [(False, 1, 10)]),
+                          ("channel", '2:9', [(True, 2, 9)]),
+                          ("channel", '2:3,7:9', [(True, 2, 9), (False, 4, 6)]),
+                          ("channel", '1:4,7:9', [(True, 2, 9), (False, 4, 6), (True, 0, 4)]),
+                          ("channel", '2:3,5:10', [(True, 2, 9), (False, 4, 6), (True, 5, 13)]),
+                          ("channel", '', [(True, 2, 9), (False, 4, 6), (True, 5, 13), (False, 0, 13)]),
+                          ("channel", '1:10', [(True, 2, 9), (False, 4, 6), (True, 0, 13)]),
+                          # a few checks of non-integer channel limits (we don't explicitly
+                          # say what this means so just check we know what it does)
+                          ("channel", '3:7', [(True, 2.8, 7.9)]),
+                          ("channel", '3:7', [(True, 2.1, 7.2)]),
+                          ("channel", '1:2,8:10', [(False, 2.8, 7.9)]),
+                          ("energy", '0.3:2.1', []),
+                          ("energy", '', [(False, 0.3, 2.1)]),
+                          ("energy", '', [(False, 0, 3)]),
+                          ("energy", '0.5:1.9', [(True, 0.51, 1.98)]),
+                          ("energy", '0.5:1.1,1.7:1.9', [(True, 0.51, 1.98), (False, 1.24, 1.51)]),
+                          ("energy", '0.3:1.3,1.7:1.9', [(True, 0.51, 1.98), (False, 1.24, 1.51), (True, 0.001, 1.32)]),
+                          ("energy", '0.5:1.1,1.5:2.1', [(True, 0.51, 1.98), (False, 1.24, 1.51), (True, 1.46, 12.2)]),
+                          ("energy", '', [(True, 0.51, 1.98), (False, 1.24, 1.51), (True, 1.46, 12.2), (False, 0.01, 13)]),
+                          ("energy", '0.3:2.1', [(True, 0.51, 1.98), (False, 1.24, 1.51), (True, 0.01, 13)]),
+                          ("wave", '5.9:41.3', []),
+                          ("wave", '', [(False, 1, 70)]),
+                          ("wave", '6.5:24.8', [(True, 6.5, 25)]),
+                          ("wave", '6.5:8.3,13.8:24.8', [(True, 6.5, 25), (False, 9.1, 12)]),
+                          ("wave", '5.9:9.5,13.8:24.8', [(True, 6.5, 25), (False, 9.1, 12), (True, 1, 10)]),
+                          ("wave", '6.5:8.3,11.3:41.3', [(True, 6.5, 25), (False, 9.1, 12), (True, 12, 70)]),
+                          ("wave", '5.9:41.3', [(True, 6.5, 25), (False, 9.1, 12), (True, 1, 70)]),
```

Comment:
There are some tests added in later PRs (not in this particular test though I think).

Reply:
You could just repeat two or three of these here with a `None` instead of the max/min values, e.g. 
`("wave", '', [(False, None, 70)])`, `("wave", '', [(False, 0, None)])`